libraries/AP_Scripting/AP_Scripting.cpp

```diff
@@ -206,6 +206,32 @@ void AP_Scripting::thread(void) {
     gcs().send_text(MAV_SEVERITY_CRITICAL, "Scripting has stopped");
 }
 
+void AP_Scripting::handle_mission_command(const AP_Mission::Mission_Command& cmd_in)
+{
+    if (!_enable) {
+        return;
+    }
+
+    WITH_SEMAPHORE(mission_data.sem);
```

Comment:
You should scope this tighter. You only need the semaphore around the creation of the mission buffer, for the actual push into the queue you don't need to hold the semaphore. Arguably you almost don't need a semaphore here at all, but I'm not actually sure that setting a pointer is actually atomic. (It almost certainly normally is in practice, but I don't know if it's guaranteed)

Reply:
Er better question: This is the only place you grab the semaphore, if no one else uses this, and multiple threads don't start a command (they don't) then I'm not sure why it's here?

Reply:
removed the semaphore